cmd/collector/app/handler/zipkin_receiver.go

```diff
@@ -25,6 +29,66 @@ var (
 	zipkinID            = component.NewID(zipkinComponentType)
 )
 
+// zipkinReceiverWrapper wraps the Opentelemetry zipkin receiver to apply keep alive settings
+type zipkinReceiverWrapper struct {
+	receiver.Traces
+	keepAlive bool
+	logger    *zap.Logger
+}
+
+// start wraps the original start method to apply keep-alive settings
+func (w *zipkinReceiverWrapper) Start(ctx context.Context, host component.Host) error {
+	err := w.Traces.Start(ctx, host)
+	if err != nil {
+		return err
+	}
+
+	if !w.keepAlive {
+		if err := w.disableKeepAlive(); err != nil {
+			w.logger.Warn("Failed to disable keep-alive on Zipkin receiver", zap.Error(err))
+		} else {
+			w.logger.Info("Disabled keep-alive on Zipkin receiver")
+		}
+	}
+
+	return nil
+}
+
+// disableKeepAlive use reflection and unsafe operations to access the internal HTTP server and disable keep alive
+func (w *zipkinReceiverWrapper) disableKeepAlive() error {
+	if w.Traces == nil {
+		return errors.New("receiver is nil")
+	}
+
+	receiverValue := reflect.ValueOf(w.Traces)
```

Comment:
it seems like you're trying to unwrap the zipkin receiver internals and set a flag. Why not just create a PR in OTEL contrib to expose the keepalive (assuming it's not exposed now)?